commit d63ff35c252256af6637cea343f7d2cfec3f36d0
Author: Cody Peter Mello <cody.mello@joyent.com>
Date:   2019-07-02T20:38:33+00:00 (3 months ago)
    
    TRITON-1196 Update FWAPI to node v6
    Reviewed by: Chris Burroughs <chris.burroughs@joyent.com>
    Approved by: Chris Burroughs <chris.burroughs@joyent.com>

diff --git a/Makefile b/Makefile
index 547efe7..913f13c 100644
--- a/Makefile
+++ b/Makefile
@@ -43,15 +43,16 @@ JSSTYLE_FLAGS    = -o indent=2,doxygen,unparenthesized-return=0,strict-indent=tr
 SMF_MANIFESTS_IN = smf/manifests/fwapi.xml.in
  
 ifeq ($(shell uname -s),SunOS)
-	# sdc-*-multiarch 15.4.1.
-	NODE_PREBUILT_IMAGE=18b094b0-eb01-11e5-80c1-175dac7ddf02
-	NODE_PREBUILT_VERSION=v0.10.48
-	NODE_PREBUILT_TAG=zone
+	# triton-origin-x86_64 19.1.0
+	NODE_PREBUILT_IMAGE=fbda7200-57e7-11e9-bb3a-8b0b548fcc37
+	NODE_PREBUILT_VERSION=v6.17.0
+	NODE_PREBUILT_TAG=zone64
 	include ./deps/eng/tools/mk/Makefile.node_prebuilt.defs
 	include ./deps/eng/tools/mk/Makefile.agent_prebuilt.defs
 else
-	NPM_EXEC :=
-	NPM = npm
+	NODE := node
+	NPM = $(shell which npm)
+	NPM_EXEC := $(NPM)
 endif
 include ./deps/eng/tools/mk/Makefile.smf.defs
 
@@ -61,7 +62,7 @@ RELEASE_TARBALL := $(NAME)-pkg-$(STAMP).tar.gz
 PKGDIR          := $(TOP)/$(BUILD)/pkg
 INSTDIR         := $(PKGDIR)/root/opt/smartdc/fwapi
 
-BASE_IMAGE_UUID = 04a48d7d-6bb5-4e83-8c3b-e60a99e0f48f
+BASE_IMAGE_UUID = cbf116a0-43a5-447c-ad8c-8fa57787351c
 BUILDIMAGE_NAME = $(NAME)
 BUILDIMAGE_DESC	= SDC FWAPI
 AGENTS		= amon config registrar
@@ -84,7 +85,7 @@ $(FAUCET): | $(NPM_EXEC)
 
 .PHONY: test
 test: $(ISTANBUL) $(FAUCET)
-	@$(ISTANBUL) cover --print none test/unit/run.js | $(FAUCET)
+	@$(NODE) $(ISTANBUL) cover --print none test/unit/run.js | $(FAUCET)
 
 .PHONY: teststop
 teststop:
diff --git a/package.json b/package.json
index 6555450..ccb3325 100644
--- a/package.json
+++ b/package.json
@@ -8,32 +8,32 @@
     "moray-sandbox": "0.2.1",
     "faucet": "0.0.1",
     "istanbul": "0.4.5",
-    "eslint": "2.13.1",
-    "eslint-plugin-joyent": "~1.3.0",
+    "eslint": "4.19.1",
+    "eslint-plugin-joyent": "~2.0.0",
     "mockery": "1.4.0"
   },
   "dependencies": {
     "assert-plus": "1.0.0",
-    "bunyan": "1.8.10",
+    "bunyan": "1.8.12",
     "clone": "0.1",
     "cmdln": "1.1.1",
     "effluent-logger": "git+https://github.com/joshwilsdon/effluent-logger.git#d662f161a07f94045ad2afb45442931511c40e51",
     "extsprintf": "1.0.2",
-    "fast-messages": "1.0.1",
+    "fast-messages": "2.0.0",
     "fwrule": "2.1.0",
     "jsprim": "^2.0.0",
     "moray": "^3.5.0",
     "moray-filter": "1.0.0",
     "tape": "4.2.2",
-    "sdc-clients": "9.5.0",
+    "sdc-clients": "12.2.0",
     "ufds": "1.2.0",
     "uuid": "3.2.1",
-    "restify": "4.3.0",
+    "restify": "4.3.4",
     "restify-warden": "0.2.0",
     "trace-event": "1.3.0",
     "triton-netconfig": "1.3.0",
     "vasync": "1.6.4",
-    "verror": "1.9.0"
+    "verror": "1.10.0"
   },
   "license": "MPL-2.0"
 }
